java/core/src/java/org/apache/orc/impl/RecordReaderImpl.java

```diff
@@ -462,8 +483,30 @@ public long getNext() {
       this(predicate, null, null, hasNulls, false, false);
     }
 
+    ValueRange(PredicateLeaf predicate, boolean hasNulls, boolean hasValue, boolean comparable) {
```

Comment:
Shall we remove this addition? I believe we can use the other constructor instead of adding this short cut.

Reply:
You're right. The previous place already uses the full constructor, so this addition is redundant and I have removed it